Common/Securities/CashBook.cs

```diff
@@ -408,5 +409,39 @@ protected virtual void OnCashAdded(Cash cash)
             var handler = CashAdded;
             handler?.Invoke(this, cash);
         }
+
+        private bool Remove(string symbol, bool calledInternally)
+        {
+            Cash cash = null;
+            var removed = _currencies.TryRemove(symbol, out cash);
+            if (!removed)
+            {
+                if (!calledInternally)
+                {
+                    Log.Error($"CashBook.Remove(): Failed to remove the cash book record for symbol {symbol}");
+                }
+            }
+            else
+            {
+                cash.Updated -= OnCashUpdate;
+                if (!calledInternally)
+                {
+                    OnUpdate();
+                }
+            }
+            return removed;
+        }
+
+        private void OnCashUpdate(object sender, EventArgs eventArgs)
+        {
+            OnUpdate();
+        }
+
+        private void OnUpdate()
+        {
+            // copy for thread safety
+            var handler = Updated;
+            handler?.Invoke(this, EventArgs.Empty);
```

Comment:
I think the copy is no longer necessary when using the `?.` operator:
https://codeblog.jonskeet.uk/2015/01/30/clean-event-handlers-invocation-with-c-6/

`Updated?.Invoke(this, EventArgs.Empty);` should be safe

Reply:
Right but since we are using `.Net 4.5.2` I think we are using C# 5? 
https://www.tutorialsteacher.com/csharp/csharp-version-history (I wasn't able to quickly find a more official link 😅 )

Reply:
All Lean projects are configured to build with C# 6 compiler (which is supported with VS 2015 and later versions). We have NET 4.5.2 as target framework version, but we could build with any C# compiler version, also C#7 which have new syntax features, but this would require at least VS 2017.

Reply:
Awesome, I'll update this then 👍